index.bs

```diff
@@ -8041,38 +8043,32 @@ ECMAScript Object values.
 
 <h4 id="es-promise">Promise types — Promise&lt;|T|&gt;</h4>
 
-IDL [=promise type=] values are
-represented by ECMAScript {{ECMAScript/Promise}}
-objects.
+IDL [=promise type=] values are represented by ECMAScript {{ECMAScript/PromiseCapability}} records.
 
 <div id="es-to-promise" algorithm="convert an ECMAScript value to promise">
 
     An ECMAScript value |V| is [=converted to an IDL value|converted=]
     to an IDL <a interface lt="Promise">Promise<var ignore>T</var></a> value as follows:
 
-    1.  Let |promise| be [=!=] <a abstract-op>Call</a>({{%Promise_resolve%}}, {{%Promise%}}, «|V|»).
-    1.  Return the IDL [=promise type=] value that is a reference to the
-        same object as |promise|.
+    1.  Let |promiseCapability| be ? [$NewPromiseCapability$]({{%Promise%}}).
+    1.  Perform [=?=] [$Call$](|promiseCapability|.\[[Resolve]], <emu-val>undefined</emu-val>, « |V| »).
+    1.  Return |promiseCapability|.
 </div>
 
 <p id="promise-to-es">
     The result of [=converted to an ECMAScript value|converting=]
     an IDL [=promise type=] value to an ECMAScript
-    value is the {{ECMAScript/Promise}} value that represents a reference to the same object that the
+    value is the value of the \[[Promise]] field of the record that
     IDL [=promise type=] represents.
 </p>
 
 <div algorithm>
 
-    One can <dfn id="dfn-perform-steps-once-promise-is-settled" export lt="upon settling">perform some steps once a promise is settled</dfn>.
-    There can be one or two sets of steps to perform,
-    covering when the promise is fulfilled, rejected, or both.
-    When a specification says to perform some steps once a promise is settled,
-    the following steps must be followed:
+    To <dfn id="dfn-perform-steps-once-promise-is-settled" export lt="upon settling">perform some steps once a promise is settled</dfn>,
+    given a <a interface lt="Promise">Promise&lt;|T|&gt;</a> |promise| and one or two sets of steps to perform,
+    covering when the promise is fulfilled, rejected, or both, perform the following steps:
 
-    1.  Let |promise| be the promise object of type <a interface lt="Promise">Promise&lt;|T|&gt;</a>.
-    1.  Let |onFulfilled| be a new [=built-in function object=] whose
-        behavior when invoked is as follows:
+    1.  Let |onFulfilledSteps| be the following steps:
```

Comment:
I think "given an argument |V|" would be fine here and remove some of the future boilerplate. (Same for onRejected.)